doublestar.go

```diff
@@ -1,8 +1,13 @@
 package doublestar
 
 import (
+	"errors"
 	"path"
 )
 
 // ErrBadPattern indicates a pattern was malformed.
 var ErrBadPattern = path.ErrBadPattern
+
+// ErrPatternNotExist indicates that the pattern passed to Glob, GlobWalk, or
+// FilepathGlob references a path that does not exist.
+var ErrPatternNotExist = errors.New("pattern does not exist")
```

Comment:
```suggestion
var ErrPathNotExist = errors.New("path does not exist")
```

This would be more exact.

Reply:
Thanks for the feedback! While I agree that `PatternNotExist` sounds weird, my first thought when I see `PathNotExist` is "what path"? I think `PatternNotExist` is more descriptive of what input caused the error since Glob's argument is called `pattern`.